### ISSUE TITLE
Bug: Can't use createRoot with UMD without "react-dom/client" warning

### ISSUE THREAD

**Issue by kentcdodds** [facebook/react#24372]
React version: v18.0.0

## Steps To Reproduce

Run this:

```html
<!DOCTYPE html>
<html>
<body>
  <div id="root"></div>
  <script src="https://unpkg.com/react@18.0.0/umd/react.development.js"></script>
  <script src="https://unpkg.com/react-dom@18.0.0/umd/react-dom.development.js"></script>
  <script type="module">
    const rootElement = document.getElementById('root')
    const element = React.createElement('div', {
      className: 'container',
      children: 'Hello World',
    })
    ReactDOM.createRoot(rootElement).render(element)
  </script>
</body>
</html>
```

Link to code example: https://jsbin.com/hucetewebu/edit?html,js,console,output

## The current behavior

Getting a console error:

> index.js:1 Warning: You are importing createRoot from "react-dom" which is not supported. You should instead import it from "react-dom/client".

However, there's no UMD version of `react-dom/client` so the error is not actionable.

I am able to work around this by adding this to the top of the script:

```js
ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.usingClientEntryPoint = true
```

But obviously that's not a great solution, especially for what I'm trying to do 😅

## The expected behavior

Disable the warning for UMD or give some other way to use createRoot from the UMD.

## Use case

I use the UMD as a way to ease people into React without introducing a bunch of tooling up-front. Getting an error in the console is not a great experience.

**Comment by billyjanitsch** [comment]
Is this a duplicate of https://github.com/facebook/react/issues/24221? If so, it was fixed by https://github.com/facebook/react/pull/24274 (but not released yet).

**Comment by gaearon** [comment]
Yes, this is a fixed bug. We're hoping to release next week if everything goes well.

**Comment by kentcdodds** [comment]
Good news! Thanks!

**Comment by niemyjski** [comment]
@gaearon is this tracked by any specific issue or what release this will go out in? Currently hitting [this](https://stackoverflow.com/questions/71748077/typeerror-cannot-set-property-usingcliententrypoint-of-undefined) in CRA app.

**Comment by gaearon** [comment]
Fixed in 18.1

**Comment by jackieli123723** [comment]
18.2.0 work
